core/src/services/hdfs/backend.rs

```diff
@@ -223,10 +227,9 @@ fn tmp_file_of(path: &str) -> String {
 /// Backend for hdfs services.
 #[derive(Debug, Clone)]
 pub struct HdfsBackend {
-    pub info: Arc<AccessorInfo>,
-    pub root: String,
+    root: String,
     atomic_write_dir: Option<String>,
-    pub client: Arc<hdrs::Client>,
+    core: HdfsCore,
```

Comment:
Although this approach (wrapping Arc when needed) will work. How about using `Arc<HdfsCore>` that owns the pointer to the core? Later, when using this `HdfsCore`, we only need to do `Arc::clone` to bump Arc's counter.